sql/moz-fx-data-shared-prod/monitoring_derived/stable_table_column_counts_v1/script.sql

```diff
@@ -0,0 +1,53 @@
+DECLARE datasets ARRAY<STRING>;
+
+DECLARE i INT64 DEFAULT 0;
+
+SET datasets = (
+  SELECT
+    ARRAY_AGG(schema_name)
+  FROM
+    `moz-fx-data-shared-prod.INFORMATION_SCHEMA.SCHEMATA`
+  WHERE
+    schema_name LIKE '%_stable%'
+);
+
+CREATE TEMP TABLE
+  columns(dataset STRING, table STRING, total_columns INT64);
+
+LOOP
+  SET i = i + 1;
+
+  IF
+    i > ARRAY_LENGTH(datasets)
+  THEN
+    LEAVE;
+  END IF;
+
+  EXECUTE IMMEDIATE '''
+    INSERT columns
+    SELECT "''' || datasets[
+    ORDINAL(i)
+  ] || '''" AS dataset, 
+      table_name, 
+      COUNT(DISTINCT(field_path)) AS total_columns 
+    FROM `moz-fx-data-shared-prod`.''' || datasets[
+    ORDINAL(i)
+  ] || '''.INFORMATION_SCHEMA.COLUMN_FIELD_PATHS
+    WHERE
+      data_type NOT LIKE "%STRUCT%"
+      AND data_type NOT LIKE "%ARRAY%"
+    GROUP BY
+      table_name  
+    ''';
+END LOOP;
+
+-- Insert into target table
+INSERT INTO
+  monitoring_derived.stable_table_column_counts_v1
+SELECT
+  DATE(@submission_date) AS submission_date,
+  dataset,
+  table_name,
+  total_columns
+FROM
+  columns
```

Comment:
Question: how do scripts like this behave when you dry run them? I assume they won't do the insert but any other gotchas?

Reply:
When dry running scripts BigQuery always returns a success state without actually dry running or executing the script. So we also can't get schema information etc from scripts